### PR TITLE
Dogs become tamed when given bones

### PR DESCRIPTION
**Pull request by Jacquerel**
## About The Pull Request

Previous to this PR Ian (and any other dog) would swear allegiance and kill (or at least growl aggressively) in the name of anyone who clicked on them sufficiently with combat mode disabled, which is boring.
Now instead Ian wants Jumbo Dog Bones.
The HoP and Warden (as dog owners) both start with one of these mythical items which otherwise can be ordered from cargo if you want to get into a dog's heart, but they're not cheap.
You can also use regular bones of the kind you may harvest from Lavaland animals or Skeleton crew members but these dogs are picky and so you have significantly less chance of success than if you buy the premium package.

Jumbo Dog Bones also fully heal dogs, as a bonus.

Additionally to this, I added functionality to the `tameable` component to allow it to not delete itself upon successful taming, to preserve Ian's ability to have more than one friend.
Additionally to that, I split the "stop being friends if you attack me" behaviour into its own element rather than being part of the "start being friends if you pet me a lot" one, so he could also keep doing that.
And finally I added a new signal `COMSIG_ATOM_AFTER_ATTACKEDBY` which is sent by an atom after an item attacks it (so if it manages to get through the attack chain and is probably actually being used for an attack) and registered to this in the `relay_attackers` element in place of `COMSIG_PARENT_ATTACKBY`. This means that dogs can now recognise that when you use a bone on them you are trying to feed them, not attack them.

## Why It's Good For The Game

The loyalty of Ian and McGriff will now be most easily available to their owners rather than anyone with access to their office.
... which doesn't really effect very much, I just think spam-clicking Ian was boring as a solution and spending money is at least marginally better.
Gives you something to spend money on if you want to befriend a dog, for instance if you're a mailperson or felinid who is tired of getting growled at by one. Or if you bought a dog from cargo.
Removes some unintuitive edge cases where a mob might get mad at you doing something helpful to them.

## Changelog

:cl:
balance: While Ian still enjoys being petted, his heart will now most closely belong to anyone who gives him an expensive luxury dog bone.
code: Using an item which _can_ deal damage on a mob tracking hostile actions (such as a dog) won't automatically be considered to be a hostile action if you aren't actually hitting it with the item.
/:cl:

### CODE REVIEW

File: code/__DEFINES/dcs/signals/signals_atom/signals_atom_attack.dm [hunk 1]
```diff
@@ -6,6 +6,8 @@
 #define COMSIG_PARENT_ATTACKBY "atom_attackby"
 /// From base of [atom/proc/attacby_secondary()]: (/obj/item/weapon, /mob/user, params)
 #define COMSIG_PARENT_ATTACKBY_SECONDARY "atom_attackby_secondary"
+///from [/item/afterattack()] called on the item which was attacked: (/obj/item/weapon, /mob/user, proximity_flag, click_parameters)
```
Comment (Fikou) [inline code/__DEFINES/dcs/signals/signals_atom/signals_atom_attack.dm:9]: dont you call it on the target though?

Comment (Jacquerel) [reply code/__DEFINES/dcs/signals/signals_atom/signals_atom_attack.dm:9]: This comment could definitely be clearer about what I mean

**Comment by Bawhoppen** [comment]
What a wonderful PR

**Comment by SomeoneYouProbablyKnow** [comment]
Cute. But i think now instead of being able to tame and command a dog you have to do a lot of extra work, which means that this will simply happen much less often (with anyone who is not HOP or Warden)

**Comment by calliophis-maculiceps** [comment]
this is not emergent gameplay

**Comment by Bawhoppen** [comment]
it's emergent code

**Comment by SomeoneYouProbablyKnow** [comment]
I mean if you want to limit it then this is an opportunity to expand on what dogs can do. If you have to get an item to feed them then let them pack a punch in combat or maybe let them sniff people out or something. But i am idea guyposting

**Comment by ExcessiveUseOfCobblestone** [comment]
The bone thing is cute but why you're punished for not buying a bone when there are creative ways to obtain one just for a dog to like you seems a bit overkill

**Comment by Jacquerel** [comment]
> The bone thing is cute but why you're punished for not buying a bone when there are creative ways to obtain one just for a dog to like you seems a bit overkill

Yeah on reflection there's no need for the chance to be tiny, I'll boost it up.
The cargo one is still going to be the only 100% chance though because bones on lavaland drop in stacks of several at once.

**Pull request merged**